tools/debug-ui/src/CurrentPeersView.tsx

```diff
@@ -0,0 +1,305 @@
+                if (oneEpoch.epoch_id === epochId) {
+                    blockProducers = new Set(
+                        oneEpoch.block_producers.map(bp => bp.account_id));
+                    chunkProducers = new Set(oneEpoch.chunk_only_producers);
+                    break;
+                }
+            }
+
+            const currentHeight = fullStatus.sync_info.latest_block_height;
+            const numPeersByStatus = new Map<string, number>();
+            const peers = [] as PeerInfo[];
+            for (const peer of networkInfo.connected_peers) {
+                const validator = [];
+                const routedValidator = [];
+                for (const element of networkInfo.known_producers) {
+                    if (blockProducers.has(element.account_id) || chunkProducers.has(element.account_id)) {
+                        if (element.peer_id === peer.peer_id) {
+                            // This means that the peer that we're connected to is a validator.
+                            validator.push(element.account_id);
+                        } else if (element.next_hops?.includes(peer.peer_id)) {
+                            // This means that the peer that we're connected to is on the shortest path
+                            // to this validator.
+                            routedValidator.push(element.account_id);
+                        }
+                    }
+                }
+                const statusClassName = peerClass(currentHeight, peer.height || 0);
+                numPeersByStatus.set(statusClassName, (numPeersByStatus.get(statusClassName) || 0) + 1);
+                peers.push({
+                    peer,
+                    validator,
+                    routedValidator,
+                    statusClassName,
+                });
+            }
+            return {
+                blockProducers,
+                chunkProducers,
+                knownSet,
+                reachableSet,
+                numPeersByStatus,
+                peers,
+            }
+        }
+        return {
+            blockProducers: new Set(),
+            chunkProducers: new Set(),
+            knownSet: new Set(),
+            reachableSet: new Set(),
+            numPeersByStatus: new Map(),
+            peers: [],
+        };
+    }, [fullStatus, epochInfo]);
+
+    if (fullStatusLoading || epochInfoLoading) {
+        return <div>Loading...</div>;
+    }
+    if (fullStatusError || epochInfoError) {
+        return <div className="network-info-view">
+            <div className="error">
+                {((fullStatusError || epochInfoError) as Error).stack}
+            </div>
+        </div>;
+    }
+    if (!fullStatus || !epochInfo) {
+        return <div className="network-info-view">
+            <div className="error">No Data</div>
+        </div>;
+    }
+
+    const detailedDebugStatus = fullStatus.detailed_debug_status!;
+
+    return <div className="current-peers-view">
```

Comment:
We also display the node's PeerId at the top now: https://github.com/near/nearcore/blob/master/chain/jsonrpc/res/network_info.html#L227